server/src/main/java/org/elasticsearch/action/support/replication/TransportBroadcastReplicationAction.java

```diff
@@ -99,7 +101,8 @@ public void accept(ActionListener<Response> listener) {
 
                 final ClusterState clusterState = clusterService.state();
                 final ProjectMetadata project = projectResolver.getProjectMetadata(clusterState);
```

Comment:
nit: I don't know how expensive a projectResolver is but perhaps it's a little cheaper to resolve once and then retrieve ProjectMetadata from projectState.metadata() ?